### DISCUSSION TITLE
Welcome to Encryption-and-Licensing Discussions!

### DISCUSSION THREAD

**Discussion by hubertus65** [hubertus65/SEMLA#7] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.
  * Post meeting minutes form the working group publicly so that they are availabel to read and comment for all members

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by hubertus65** [comment]
# Lunch Meeting at Modelica conference, October 10th 2023

Maplesoft, Elena: positive to overall initiative, but no time to implement this currently.

Dassault, Dan: Prototype available, maybe updates necessary, more tech detail needed in specification, e.g., for the manifest. We need use cases (Peter Lundberg, Dassault, is the right contact), and an updated OpenSSL.

TLK, Henrik, positive since standardized for all libraries, some improved documentation, had trouble getting it to work. Current testing tool in the repository works for one unit test: better testing infrastructure is needed. For example, a test package for encryption and decryption. We had no troubles implementing this. CI-pipeline issues with CONAN configuration.

XRG, Robert Flesch: Please provide documentation updates. We have a working solution working for OpenModelica and Modelon Impact, and also would like better testing infrastructure,

OpenModelica, Adeel, Adrian and Martin S: Have done a prototype implementation for Bosch Rexroth, would like more documentation, especially how library developers can work with this. Testing tool is very basic.

Some people want to use encryption that expires in time as a new feature.

Wolfram Mathcore, Henrik: We would like to get started soon. The Modelica "Protection and Licensing" annotations are already in place for us, we would like to use SEMLA to enforce it.

**Suggestion:** Let's add the specification for SEMLA to the "Protection and Licensing Annotations" section in the MLS.

Hans: Library Vendors should have a finished open-source solution that works with a set of tools.

Needed: an open-source reference implementation that works with different tools, and one exemplary encryption solution.

Modelon, Hubertus: Suggestion to add package management into this effort for Modelica Libraries. Need semantic versioning of Libraries for this to work. How to handle optional dependencies? For binary dependencies, use CMAKE.

**Decision**: Let's get the status of SEMLA into the standard as quickly as we manage before adding new features, but let's make sure suggested new features are considered in that design.

**Question**: Where do we get public keys? Siemens was asking for it, how should keys be distributed?

- [ ] **Action Hubertus**: call for a working meeting this fall, including Modelon's team on our side.

**Question**: Could a lib vendors put their implementation into the repo?

**Reply by hubertus65** [reply]
All of you who were there: please complement with whatever I forgot! I will share the link to discussions with our mailing list.

**Comment by d94pn** [comment]
Note the issues reported here: 

https://github.com/modelica/Encryption-and-Licensing/issues

that were discovered during implementation.

**Comment by niklwors** [comment]
My name is Niklas Worschech and I am working for Bosch Rexroth.  We strongly support the  standardisation and we are very happy that we have the opportunity to do this with SEMLA. We would like to licence and encrypt our **Rexroth BRSL** library with SEMLA. We have already done it for OpenModelica and released our BRSL 3.0 for OMEdit 1.22.0. 
We encountered a few problems for which I have also created tickets here:
https://github.com/modelica/Encryption-and-Licensing/issues
